lit/docs/guides/pipeline-guides.lit

```diff
@@ -0,0 +1,676 @@
+\title{Pipeline Guides}
+
+\use-plugin{concourse-docs}
+\split-sections
+\table-of-contents
+
+\section{
+  \title{Common Pipeline Practices}
+  \omit-children-from-table-of-contents
+
+  The following are practices that we see a lot of people use in their
+  pipelines. These are by no means "Best" practices, they are simply common and
+  may or may not work for you and your team.
+
+  \section{
+    \title{Parallelizing Get Steps in Jobs}
+
+    All jobs usually have \reference{get-step}{get steps} as their first set of
+    steps.
+
+    \codeblock{yaml}{{
+    jobs:
+    - name: awesome-job
+      plan:
+      - get: cool-code
+      - get: funny-binary
+      - get: the-weather
+      - task: business-stuff
+    }}
+
+    To reduce the waiting time to the length of longest running get step, put all
+    get steps under an \reference{in-parallel-step}.
+
+    \codeblock{yaml}{{
+    jobs:
+    - name: awesome-job
+      plan:
+      - in_parallel:
+        - get: cool-code
+        - get: funny-binary
+        - get: the-weather
+      - task: business-stuff
+    }}
+  }
+
+  \section{
+    \title{Specify Inputs for Put Steps}
+
+    By default \reference{put-step}'s have all artifacts from a job mounted in
+    their resource container. This can result in long initialization times for
+    put steps. It's likely that a \reference{put-step} only needs a subset of
+    all available artifacts generated throughout the job.
+
+    There are two ways to specify which artifacts to send to a
+    \reference{put-step}. You can specify \code{detect} as the
+    \reference{schema.step.put-step.inputs} or you can pass in an exact list of
+    all artifacts the \reference{put-step} needs.
+
+    Using \code{detect}:
+    \codeblock{yaml}{{
+    jobs:
+    - name: the-job
+      plan: # Get some artifacts
+      - get: apples
```

Comment:
minor thing, but it's a bit awkward to suggest that you typically want to wrap your `get` steps in an `in_parallel`, but not follow that advice in later examples

Reply:
Had this thought too as I was writing it 😅 I ended up not adding `in_parallel` because I wanted to keep the example focused on the put step farther down. That all said, I think I'm overthinking it, might as well follow recommended practices everywhere!